### PR TITLE
NYOSD-84 Add new endpoint for event list from NemDeling

### PR DESCRIPTION
**Pull request by googletorp**

### CODE REVIEW

File: src/nemdeling/nemdeling.api-docs.rest [hunk 1]
```diff
@@ -102,3 +102,62 @@ Content-Type: application/xml
...
+                <img src="https://union.kk.dk/sites/default/files/2022-08/IMG_20220507_0036%201920x1080.jpg" alt="Yacarandá besøger Union 21. oktober 2022" height="1080" width="1920" title="" />
+            </item>
+        </billede>
+        <title>Yacarandá (BRA/BOL/ARG) at Union</title>
+        <field_teaser>Elegante og skrøbelige melodier, der blander brasiliansk folklore, klassisk musik og afrobrasilianske rytmer.</field_teaser>
+        <screen is_array="true">
+            <item>copenhagen_test</item>
+            <item>screen_3</item>
+            <item>screen_4</item>
+        </screen>
```
Comment (kasperg) [inline src/nemdeling/nemdeling.api-docs.rest:133]: I see we are planning to get individual items per screen :+1:

File: src/nemdeling/nemdeling.controller.ts [hunk 2]
```diff
@@ -204,6 +209,77 @@ export class NemDelingController {
...
+    // don't want this to get completely stuck.
+    if (
+      this.nemDelingService.eventListsAreSyncing > 0 &&
+      this.nemDelingService.eventListsAreSyncing < 5
+    ) {
+      this.nemDelingService.eventListsAreSyncing++;
+      throw new ServiceUnavailableException("Event lists are already being synced.");
+    }
+
+    this.nemDelingService.eventListsAreSyncing = 1;
```
Comment (kasperg) [inline src/nemdeling/nemdeling.controller.ts:247]: I appreciate the comment but this seems like a somewhat primitive approach to handling synchronicity which is now duplicated three times in the controller. I wonder if you have considered other alternatives like transactions, a semaphore or a mutex.

File: src/nemdeling/nemdeling.controller.ts [hunk 3]
```diff
@@ -247,14 +323,18 @@ export class NemDelingController {
           backgroundColor = item.background_color[0];
         }
 
+        const [startTime, endTime] = item.time[0].item[0].split(" til ");
+        const startDate = this.nemDelingService.formatEventDate(item.startdate[0].item[0]);
+        const endDate = this.nemDelingService.formatEventDate(item.enddate[0].item[0]);
+
```
Comment (kasperg) [inline src/nemdeling/nemdeling.controller.ts]: Consider moving this processing outside this `forEach`. It does not seem to be related to the individual screen.

File: src/nemdeling/nemdeling.controller.ts [hunk 4]
```diff
@@ -247,14 +323,18 @@ export class NemDelingController {
...
         result[screenName].push({
           templateId,
           content: {
             title: item.title[0],
             subTitle: item.field_teaser[0],
             host: item.host[0],
-            date: this.nemDelingService.formatEventDate(item.startdate[0].item[0]),
-            time: `kl. ${item.time[0].item[0]}`,
+            startDate: `${startDate} kl. ${startTime}`,
+            endDate: startDate !== endDate ? `${endDate} kl. ${endTime}` : "",
```
Comment (kasperg) [inline src/nemdeling/nemdeling.controller.ts:337]: As I read the code this means we will not display the end time if this is a single day event. We did not do so before either so perhaps that is fine.

File: src/nemdeling/nemdeling.service.ts [hunk 5]
```diff
@@ -31,6 +31,15 @@ export class NemDelingService {
    */
   public eventsAreSyncing = 0;
 
+  /**
+   * If event lists are syncing.
+   *
+   * 0 means that no sync is in progress.
+   * A value greater than 0 means that syncing is in progress, the number is the
+   * number og requests that has been skipped.
```
Comment (kasperg) [inline src/nemdeling/nemdeling.service.ts]:
```suggestion
   * number of requests that has been skipped.
```

**Review by kasperg — APPROVED** [review]
:+1: This follows the existing approach in the project nicely.

I have some comments you can look in to of disregard at your preference.

File: src/nemdeling/nemdeling.controller.ts (hunk 2 above)
Comment (googletorp) [reply src/nemdeling/nemdeling.controller.ts:247]: I wanted to get some real world experience before doing something a bit more advanced or maybe remove it completely. What are you doing in the old version today?

File: src/nemdeling/nemdeling.controller.ts (hunk 4 above)
Comment (googletorp) [reply src/nemdeling/nemdeling.controller.ts:337]: Yes, that's the current request, to only display end time/date when end date is different than start date
